### DISCUSSION TITLE
How to ignore unknown Elements?

### DISCUSSION THREAD

**Discussion by EMaksymenko** [pdvrieze/xmlutil#96] [Q&A] [answered]
I am investigating a library code and could not understand how to configure XML parser to ignore unknown Elements in XML file?

Could anybody write a code example to do so?

I am having XML file which contains additional name-spaces with extensions not described in basic XSD.

Is it a correct way?
```
    @OptIn(ExperimentalXmlUtilApi::class)
    protected val xml = XML {
        unknownChildHandler = UnknownChildHandler { _, _, _, _, _ -> emptyList() }
    }
```

**Comment by pdvrieze** [comment, marked as answer]
Your example is the intended way to support ignoring unknown elements/namespaces.

**Reply by pdvrieze** [reply]
Please note that as of `0.85.0` it is possible to get this behaviour through the builder for the default policy.

**Comment by p-fischer** [comment]
Usually I don't want my client to break if a new XML tag is added in an endpoint. 
Wouldn't the above implementation be a good candidate for the default `UnknownChildHandler`?
... Or to not break the API, this could be exposed as `GRACEFUL/RECOVERABLE_UNKNOWN_CHILD_HANDLER`.

**Reply by pdvrieze** [reply]
That is an idea. I can certainly provide some default alternatives (ignore all unknown children, ignore children in different namespaces,...). I might even add a shortcut.
